packages/drivers/odsp-driver/src/odspDocumentService.ts

```diff
@@ -504,55 +223,7 @@ export class OdspDocumentService implements IDocumentService {
         // In such case client cached info is stale and has to be removed.
         if (error !== undefined) {
             this.epochTracker.removeEntries().catch(() => {});
-        } else {
-            this._opsCache?.flushOps();
         }
-        this._opsCache?.dispose();
-        this.clearJoinSessionTimer();
-        this.currentConnection?.dispose();
-        this.currentConnection = undefined;
-    }
-
-    protected get opsCache() {
-        if (this._opsCache) {
-            return this._opsCache;
-        }
-
-        const seqNumber = this.storageManager?.snapshotSequenceNumber;
-        const batchSize = this.hostPolicy.opsCaching?.batchSize ?? 100;
-        if (seqNumber === undefined || batchSize < 1) {
-            return;
-        }
-
-        const opsKey: Omit<IEntry, "key"> = {
-            type: "ops",
-        };
-        this._opsCache = new OpsCache(
-            seqNumber,
-            this.mc.logger,
-            // ICache
-            {
-                write: async (key: string, opsData: string) => {
-                    return this.cache.persistedCache.put({ ...opsKey, key }, opsData);
-                },
-                read: async (key: string) => this.cache.persistedCache.get({ ...opsKey, key }),
-                remove: () => { this.cache.persistedCache.removeEntries().catch(() => {}); },
-            },
-            batchSize,
-            this.hostPolicy.opsCaching?.timerGranularity ?? 5000,
-            this.hostPolicy.opsCaching?.totalOpsToCache ?? 5000,
-        );
-        return this._opsCache;
-    }
-
-    // Called whenever re receive ops through any channel for this document (snapshot, delta connection, delta storage)
-    // We use it to notify caching layer of how stale is snapshot stored in cache.
-    protected opsReceived(ops: ISequencedDocumentMessage[]) {
-        // No need for two clients to save same ops
-        if (ops.length === 0 || this.odspResolvedUrl.summarizer) {
-            return;
-        }
-
-        this.opsCache?.addOps(ops);
+        this.odspDocumentServiceDelayLoaded?.dispose(error);
```

Comment:
It will result in loading this layer, even if we did not load it before, right? Can it be avoided? I.e. do we have to call dispose to those elements, if they were never loaded?
And if we go that path, how do we ensure that accessing this.odspDocumentServiceDelayLoaded after dispose is rejected? I think we should load any code after dispose - either by having assert (and ensuring we enforce it somehow), or better maybe - by ensuring that returns some dummy object that does nothing (but implements same interface). We can start with one flow and move to another once we have data (i.e. maybe start with simple assert, and learn if we are hitting it, and if it's better to push requirement to callers, or implement dummy object)

Reply:
Removed this code and added a separate socket bundle: https://github.com/microsoft/FluidFramework/pull/13519